code/game/mecha/mecha.dm

```diff
@@ -2084,6 +2088,38 @@
 	return "huddiagmax"
 
 
+/obj/item/device/mech_painter
+	name = "Mecha Painter"
+	desc = "A device used to paint floors in various colours and fashions."
+	icon = 'icons/obj/RCD.dmi'
+	icon_state = "rpd"//placeholder art, someone please sprite it
+	force = 0
+
+/obj/item/device/mech_painter/afterattack(var/obj/mecha/M, var/mob/user)
+	if (!M.paintable)
+		to_chat(user, "<span class='info'>This mech cannot be painted.</span>")
+		return 1
+	if (!M.mech_sprites.len)
+		to_chat(user, "<span class='info'>This mech has no other paint-jobs.</span>")
+		return 1
+	if (M.occupant)
+		to_chat(user, "<span class='info'>This mech has an occupant. It must be empty before you can paint it.</span>")
+		return 1
+
+	var/icontype = input("Select the paint-job!")in M.mech_sprites
+
+	if(icontype == M.initial_icon)
+		to_chat(user, "<span class='info'>This mech is already painted in that style.</span>")
+		return 1
+	if(icontype)
+		to_chat(user, "<span class='info'>You begin repainting the mech.</span>")
+		spawn(60)
+			M.initial_icon = icontype
+			M.icon_state = icontype +"-open"
```

Comment:
instead of hard-setting as open, you should call update_icon here since someone might have started climbing in while it was being painted

Reply:
I tested this and it doesn't quite work since (I think) mech code works by providing an extension to the current icon name via proc. If I try changing it in the middle of a random state there's nothing for the code to know that it's currently in a special state unless I hard code it (I think).

Reply:
mecha update_icon code is probably really bad if what you say is true